EIPS/eip-3000.md

```diff
@@ -0,0 +1,153 @@
+---
+eip: 3000
+title: Staking Reward Calculation
+author: Tony Carson <tony.carsonn@gmail.com>, Mehmet Sabir Kiraz <m.kiraz@gmail.com>, Süleyman Kardaş <skardas@gmail.com>
+type: Standards Track
+category: ERC
+status: Draft
+created: 2020-08-28
+discussions-to: https://github.com/ethereum/EIPs/issues/3000
+---
+## Simple Summary
+ERC3000 is a new standardization for on-chain calculation of staking reward.
+## Abstract
+
+Based on the product of effective collateral and time, ERC3000 calculates the reward a user can get at any time, and realize the real decentralized DeFi. Here below is the formula for the calculation of reward for a user U:
+
+ 
+<img src="https://latex.codecogs.com/svg.latex?\Large&space;Reward_U= \sum_{i=1}^n \frac{\triangle p_i}{\triangle P_i} \times \triangle G_i " title="erc3000" />
+
+where ∆p<sub>i</sub> denotes individual productivity of the user U between the consecutive block numbers t<sub>i-1</sub>  and t<sub>i</sub>, ∆P<sub>i</sub> denotes global productivity between the consecutive block numbers t<sub>i-1</sub>  and t<sub>i</sub>, and ∆G<sub>i</sub> denotes gross product between the consecutive block numbers t<sub>i-1</sub>  and t<sub>i</sub>. The formula ensures that there is no benefit in case of exiting earlier or entering later in the computation. The reward a user can get for a period is based on his total productivity during that specific time. The formula has been simplified through Solidity and generalized design to make it available across all DeFi products. 
+We note that the smart contract can be triggered for every computation of on the following events: 
+	
+- whenever the productivity of a user changes (increase/decrease), 
+- whenever a user withdraws.
+
+## Motivation
+
+One of the main drawbacks of many DeFi projects is the reward distribution mechanism within the smart contract. In fact, there are two main mechanisms are adopted so far.
+1.	Distribution of rewards is only given when all users exit the contract
+2.	The project collects on-chain data, conducts calculation off-chain, and sends the results
+to the chain before starting rewards distribution accordingly
+
+The first approach conducts all calculation in an on-chain fashion, the cycle of its rewards distribution is too long. Furthermore, users need to remove their collateral before getting the rewards, which can be harmful for their rewards. The second approach is a semi-decentralized model since the main algorithm involves an off-chain computation. Therefore, the fairness and transparency properties cannot be reflected and this can even create the investment barrier for users.
```

Comment:
I would like to see an addition to the motivation section that explains why this needs to be standardized.  Keep in mind, not all good ideas need to be standards, many good ideas can just be good ideas.